src/stumpy_core/canvas.cr

```diff
@@ -34,5 +34,36 @@ module StumpyCore
         @height == other.height &&
         @pixels == other.pixels
     end
+    
+    def paste(canvas, x, y)
+      #StumpyPNG does not handle going out of bound well so we have to be very careful with the paste
+
+      #detect left side first
+      min_x = (x < 0 ? x.abs : 0)
+      max_x = canvas.width - (x + canvas.width >= width ? x + canvas.width - width)
```

Comment:
* does not appear to be valid crystal code (there is no alternative in the ternary operator).
* in the `=` case of `x + canvas.width >= width`
   `x + canvas.width - width` is equal to 0, so using `x + canvas.width > width` would be enough

What do you think of writing it this way?

 `max_x = x + canvas.width <= width ? canvas.width : width - x`